src/tasks/CodeDeployDeployApplication/TaskOperations.ts

```diff
@@ -113,38 +114,45 @@ export class TaskOperations {
 
         let key: string
         const bundleFilename = path.basename(archiveName)
+
         if (this.taskParameters.bundlePrefix) {
             key = `${this.taskParameters.bundlePrefix}/${bundleFilename}`
         } else {
             key = bundleFilename
         }
 
-        console.log(tl.loc('UploadingBundle', archiveName, key, this.taskParameters.bucketName))
-        const fileBuffer = fs.createReadStream(archiveName)
         try {
+            console.log(tl.loc('UploadingBundle', archiveName, key, this.taskParameters.bucketName))
+
+            const readStream = fs.createReadStream(archiveName)
+
+            readStream.on('error', err => {
+                if (err) {
+                    throw err
+                }
+            })
+
             const request: S3.PutObjectRequest = {
                 Bucket: this.taskParameters.bucketName,
                 Key: key,
-                Body: fileBuffer
+                Body: readStream
             }
 
             if (this.taskParameters.filesAcl && this.taskParameters.filesAcl !== 'none') {
                 request.ACL = this.taskParameters.filesAcl
             }
 
             await this.s3Client.upload(request).promise()
```

Comment:
Sanity check based on your PR description. Will the code wait here until the transfer completes?

Reply:
Yes that's correct